admin/shell.nix

```diff
@@ -0,0 +1,46 @@
+{
+  self ? import ./. { },
+}:
+let
+  inherit (self)
+    lib
+    pkgs
+    packages
+    venv
+    ;
+
+  scripts = lib.mapAttrsToList (name: value: value) packages;
+  scriptsList = lib.pipe packages [
+    (lib.mapAttrsToList (name: value: "- ${name}"))
+    (lib.concatStringsSep "\n")
+  ];
+in
+pkgs.mkShellNoCC {
+  packages = [
+    pkgs.uv
+    venv
+    scripts
+  ];
+
+  env = {
+    # Don't create venv using uv
+    UV_NO_SYNC = "1";
+
+    # Force uv to use Python interpreter from venv
+    UV_PYTHON = "${venv}/bin/python";
+
+    # Prevent uv from downloading managed Python's
+    UV_PYTHON_DOWNLOADS = "never";
+  };
+
+  shellHook = ''
+    # Undo dependency propagation by nixpkgs.
+    unset PYTHONPATH
```

Comment:
What about setting `PYTHONPATH` to site packages provided by create venv. This would would our editor's LSP access required Python modules.

Currently, I have to set `export PYTHONPATH=/nix/store/xflk05rk1was60bckn6w44cbgwsbrbpa-scripts-default-env/lib/python3.12/site-packages/` to get my LSP working properly.